src/landscape.cpp

```diff
@@ -1266,15 +1269,15 @@ static bool FlowRiver(TileIndex spring, TileIndex begin)
 				/* We don't want the lake at the entry of the valley. */
 				lakeCenter != begin &&
 				/* We don't want lakes in the desert. */
-				(_settings_game.game_creation.landscape != LT_TROPIC || GetTropicZone(lakeCenter) != TROPICZONE_DESERT) &&
+				(_settings_game.game_creation.landscape != LT_TROPIC || _settings_game.game_creation.lakes_allowed_in_deserts == true || GetTropicZone(lakeCenter) != TROPICZONE_DESERT) &&
 				/* We only want a lake if the river is long enough. */
 				DistanceManhattan(spring, lakeCenter) > _settings_game.game_creation.min_river_length) {
 			end = lakeCenter;
 			MakeRiver(lakeCenter, Random());
 			/* Remove desert directly around the river tile. */
-			CircularTileSearch(&lakeCenter, RIVER_OFFSET_DESERT_DISTANCE, RiverModifyDesertZone, nullptr);
+			CircularTileSearch(&lakeCenter, _settings_game.game_creation.river_tropics_width, RiverModifyDesertZone, nullptr);
 			lakeCenter = end;
-			uint range = RandomRange(8) + 3;
+			uint range = RandomRange(8) + _settings_game.game_creation.lake_size;
```

Comment:
Similar to the other PR, I wonder if the RandomRange argument should also be a function of the lake size setting.

Reply:
I'll experiment and see how this affects generation.

Reply:
> 
> 
> Similar to the other PR, I wonder if the RandomRange argument should also be a function of the lake size setting.

While it didn't work well for the other PR, it worked out fine here. I've committed a new change to make the max size of the random range our new setting, changed the default value of the setting to 8 to match the original hard coded value. Now small lakes can generate as well as large lakes. Still not a fan of the shape of the lakes in larger sizes but that's a challenge for another day.